test/test_torch.py

```diff
@@ -12698,6 +12711,133 @@ def test_function_unwrap_message(self):
         self.assertRaisesRegex(RuntimeError, ' call to _th_lt',
                                lambda: torch.ones(1, dtype=torch.float) < torch.ones(1, dtype=torch.double))
 
+    @staticmethod
+    def _test_unary_op_overlap(self, sz, data, op):
+        # c is identical to a:
+        c = data[0:sz]
+        a = data[0:sz]
+        c_exp = torch.empty_like(c)
+        op(a, out=c_exp)
```

Comment:
```
c_exp = torch.empty_like(c)
op(a, out=c_exp)
```

is equivalent to:
```
c_exp = op(a)
```

Reply:
If written correctly, yes. However it is possible to screw up and dispatch them to completely different kernels.

Reply:
Oh no, I see your point :(.

That's kind of unfortunate. Can we either (1) audit the kernels here and make sure they dispatch to the same thing or (2) write a big comment here so that no one tries to mistakingly clean the code up?

Reply:
I think we need to add comment in this PR. But as global policy, I prefer if we write only _out kernel and generate regular and in-place functions automatically.